src/zenml/utils/source_utils.py

```diff
@@ -523,47 +525,52 @@ def load_source_path_class(
         source = source.split("@")[0]
 
     if import_path is not None:
-        with prepend_python_path(import_path):
+        with prepend_python_path([import_path]):
             logger.debug(
                 f"Loading class {source} with import path {import_path}"
             )
             return import_class_by_path(source)
     return import_class_by_path(source)
 
 
-def import_python_file(file_path: str) -> types.ModuleType:
-    """Imports a python file.
+def import_python_file(file_path: str, zen_root: str) -> types.ModuleType:
+    """Imports a python file in relationship to the zen root.
 
     Args:
         file_path: Path to python file that should be imported.
+        zen_root: Path to current zenml root
 
     Returns:
         imported module: Module
     """
     # Add directory of python file to PYTHONPATH so we can import it
     file_path = os.path.abspath(file_path)
-    module_name = os.path.splitext(os.path.basename(file_path))[0]
+    module_path = os.path.relpath(file_path, zen_root)
+    # module_name = os.path.splitext(os.path.basename(file_path))[0]
 
     # In case the module is already fully or partially imported and the module
     #  path is something like materializer.materializer the full path needs to
     #  be checked for in the sys.modules to avoid getting an empty namespace
     #  module
-    full_module_path = os.path.splitext(
-        os.path.relpath(file_path, os.getcwd())
-    )[0].replace(os.path.sep, ".")
+    module_name = os.path.splitext(os.path.relpath(module_path, os.getcwd()))[
+        0
+    ].replace(os.path.sep, ".")
 
-    if full_module_path not in sys.modules:
-        with prepend_python_path(os.path.dirname(file_path)):
+    if module_name in sys.modules:
+        del sys.modules[module_name]
```

Comment:
Can you maybe add a short (or long) comment here that explains in detail why the module needs to be deleted, as that is a very strange operation if someone comes access this code.